engine/src/main/java/org/n52/javaps/engine/impl/EngineImpl.java

```diff
@@ -182,20 +192,25 @@ public JobId execute(OwsCode identifier,
 
     private Result onJobCompletion(Job job) throws EngineException {
         try {
+            this.jobStatusLock.writeLock().lock();
             this.cancelers.remove(job.getJobId());
             this.resultPersistence.save(job);
             this.jobs.remove(job.getJobId());
             return this.resultPersistence.getResult(job.getJobId());
         } finally {
             job.destroy();
+            this.jobStatusLock.writeLock().unlock();
```

Comment:
If `job.destroy()` fails, the lock stays locked...

Reply:
strictly speaking, the `job.destroy()` is the problem here. But you are right. so a `try` in a `try`? =)

Reply:
Looking at the code, switching the lines should be fine..